### DISCUSSION TITLE
Labels: duplicate labels after library scan

### DISCUSSION THREAD

**Discussion by ivvv** [photoprism/photoprism#4790] [Q&A]
I use photoprism on Raspberry Pi 4B, using the latest `docker-compose.yaml` for ARM64, with `MariaDB 10.11.3` (as reported in `docker-compose logs`. The RAM is 8 GB, 

I have several duplicated labels after scanning the library. E.g. I have two labels "Beach" with different number of photos in each label. Similarly, "Car", "Flowers" and more labels are all duplicated with different number of photos.

I have checked the labels for a few photons in each duplicated label and I cannot see anything obvious, e.g. an extra space in label name or label listed twice. Nothing obvious in the settings too.

Is there a way to fix this issue without rescanning my whole library of more than 20,000 photos? Is there a guarantee that even if I do this I will not end up with the same duplicated labels?

**Comment by lastzero** [comment]
You are welcome to create a bug report once you can tell us how to reproduce your issue. Also be sure to use our report template for this. Thank you very much! ❤️

In particular, we need to know the exact build number displayed in the footer of the Settings page.

**Comment by ivvv** [comment]
---------------------------------------------------------------------------

#### 1. What is not working as documented?

I have several duplicated labels after scanning the library. E.g. I have two labels "Beach" with different number of photos in each label. Similarly, "Bottle", "Flowers",  and more labels are all duplicated with different number of photos.
I have checked the labels for a few photons in each duplicated label and I cannot see anything obvious, e.g. an extra space in label name or label listed twice. Nothing obvious in the settings too.

#### 2. How can we reproduce it?

Steps to reproduce the behaviour:

1. Import and index ~20,000 photos (all are JPEGs)
2. Check the automatically assigned labels (no manually added labels)
3. Identify those which are duplicates
4. See screenshots [here](https://drive.google.com/file/d/1eZOMNAa4pDnLTZoKhaTk1mR8zudgHx25/view?usp=share_link).

I am using the default `docker-compose..yaml` for Raspberry Pi, downloaded on 13 May 2023. Only changed the Storage folders and no other change in parameters.

#### 3. What behaviour do you expect?

I would expect the labels to be unique, no duplicates.

#### 4. What could be the cause of your problem?

I presume the database should keep unique/distinct labels, not sure why this happens.

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

I am not sure how to provide examples, I don't think sharing my 90 GB photos library is feasible. A few screenshots of the duplicated labels are available [in here](https://drive.google.com/file/d/1eZOMNAa4pDnLTZoKhaTk1mR8zudgHx25/view?usp=share_link).

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: ARMv8, PhotoPrism build [230506-9de9a3540](https://docs.photoprism.app/release-notes/)

(b) Database Type & Version: MariaDB 10.11.3

(c) Operating System Types & Versions: Raspbian GNU/Linux 11 (bullseye)

(d) Browser Types & Versions: MS Edge Version 113.0.1774.57

(e) Ad Blockers, Browser Plugins, and/or Firewall Software? No

#### 7. On what kind of device is PhotoPrism installed?

(a) Device / Processor Type: Raspberry Pi 4B

(b) Physical Memory & Swap Space in GB: 8 GB RAM, 1GB swap.

(c) Storage Type: USB disk mounted on boot.

(d) Anything else that might be helpful to know?

As I mentioned above, all labels are automatically assigned after the import/scanning. I have not added labels.

**Comment by lastzero** [comment]
I have checked all the instances I have access to and could not find any duplicate labels. Is there a chance that your database is not using the default settings from the example config files we provide?
- In some cases, for example if you change the default character set to binary unicode instead of e.g. case-insensitive unicode, this could cause the `LIKE` comparison in MariaDB to be case sensitive, leading to the behavior described.
- This could also happen in other cases, e.g. if you have changed the charset or collation manually, or a tool you use to access the database has done so without telling you?
- Would you be able to check the `labels` table specification for us to learn more? 🤔

**Comment by ivvv** [comment]
I am quite sure I did not touch the default settings in the default `docker-compose.yaml` except the part where I set up the folders. And I have not tweaked anything with character sets.

I was going to try and check the database labels but haven't got there yet, so I'd be thankful if you could provide some guidelines on how to do this. 

On the backend (browser) I cannot see anything obvious with the duplicated labels (e.g. trailing blank space, different case). So, the direct check with the database would provide some hints, I believe.
